src/jit/codegencommon.cpp

```diff
@@ -2692,11 +2692,21 @@ void CodeGen::genExitCode(BasicBlock* block)
 
 void CodeGen::genJumpToThrowHlpBlk(emitJumpKind jumpKind, SpecialCodeKind codeKind, GenTreePtr failBlk)
 {
-    if (!compiler->opts.compDbgCode)
-    {
-        /* For non-debuggable code, find and use the helper block for
-           raising the exception. The block may be shared by other trees too. */
+    /* For non-debuggable code, find and use the helper block for
+       raising the exception. The block may be shared by other trees too.
+
+       Otherwise, the code to throw the exception will be generated inline,
+       and we will jump around it in the normal non-exception case */
+    bool useExnThrowingBlock = !compiler->opts.compDbgCode;
+
+#if !FEATURE_FIXED_OUT_ARGS
+    /* Allow the use of exception throwing blocks only for EBP frame.
+       (two basic blocks may have different stack level due to push/pop) */
+    useExnThrowingBlock = useExnThrowingBlock && (compiler->rpFrameType != FT_ESP_FRAME);
+#endif
```

Comment:
I think you should be consistent with other code and use:
```
    useExnThrowingBlock = useExnThrowingBlock && isFramePointerUsed();
```

Also, please change the comment style to `//` instead of `/*` for the comments you change/add.